### ISSUE TITLE
OSMOSE support for validating existing OSM data against the NSI

### ISSUE THREAD

**Issue by matthewdarwin** [osmlab/name-suggestion-index#2998]
I opened this issue for OSMOSE to have support to validate existing OSM data against the name suggestion index.

I am creating this issue (1) for awareness and (2) to ask if there any suggestions on validation methods.  Could you please add thoughts to the OSMOSE-BACKEND gitub?

https://github.com/osm-fr/osmose-backend/issues/609

Thanks to Bryan Housel for his great presentation on this at SOTM-US on Saturday.

**Comment by 1ec5** [comment]
This is a good excuse to add some way to indicate which tags are mandatory and which may be modified on a case-by-case basis. For example, some brands would always have the same `name`, while others are expected to differ from location to location.

**Comment by matthewdarwin** [comment]
Yes, this would be very important from a validation point of view.

**Comment by bhousel** [comment]
Thanks @matthewdarwin !  Glad you enjoyed the talk 😄 

I agree that we should do a few things:
- I can work with the OSMOSE devs to make use of this index somehow
- Some of that will involve better dev documentation for the index.  I have it integrated ok with iD but using tricks that aren't written down anywhere.
- Agree with @1ec5's point that we should add some capability for defining which tags are mandatory vs which are flexible.  `name` is a good example.

**Comment by frodrigo** [comment]
Done
https://github.com/osm-fr/osmose-backend/pull/874
http://osmose.openstreetmap.fr/en/errors/?item=3130

Partial implementation on locationSet support. Osmose deals with area using iso code, and other code for subarea. Unlike NSI using JSON.

cc @nlehuby

**Comment by bhousel** [comment]
Cool!  That's great news..

 It would probably be useful to port [location-conflation](https://github.com/ideditor/location-conflation) to Python, so you can resolve the locationSets into GeoJSON shapes.

**Comment by frodrigo** [comment]
Not interested in supporting arbitrary polygon in Osmose as we only need area/sub area and geometry from OSM to define them.

**Comment by bhousel** [comment]
> Not interested in supporting arbitrary polygon in Osmose as we only need area/sub area and geometry from OSM to define them.

The NSI now tracks handful of regional brands where the GeoJSON is important.
Peoples Bank is one example:  https://nsi.guide/index.html?t=brands&k=amenity&v=bank&tt=peoples%20bank
